include/color_spinor_field.h

```diff
@@ -830,6 +867,92 @@ namespace quda
     friend class ColorSpinorParam;
   };
 
+  using ColorSpinorField_ref = std::reference_wrapper<ColorSpinorField>;
+
+  /**
+     @brief Create a std::vector containing of subset to an input set.
+     The subset view is of type std::vector<std::reference_wrapper<T>>
+     over the range [offset, offset + count).
+     @param[in] v Input vector
+     @param[in] offset Initial index of the subset
+     @param[in] upper Size of the subset
+     @return Subset vector
+  */
+  template <typename T> inline auto make_subset(T &&v, size_t offset, size_t count)
```

Comment:
I feel like this machinery is more general that for just `ColorSpinorField`, and it might be worth putting in its own header. If anything, it might be useful for both vectors of ColorSpinorField (later) and also for the coefficients in blas/multiblas. Thoughts?

Reply:
I had actually forgotten to delete `make_subset` and `make_range` as these are no longer needed.  The blas interface now has the ability to specify subsets directly using initializer list notation, e.g., 
```c++
{x.begin() + offset, x.begin() + offset + count}
```

Only the `make_set` is now needed.

Reply:
e524640ec20406f7408962765cd215020a4fc396